### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by kamsar** [BerserkerDotNet/Unicorn.VisualStudio#1]
There doesn't seem to be a license in the repo or on the VS page. Did you have one in mind?

Any interest in seeing something like your remoting stuff pulled into the Unicorn 3 core?

**Comment by BerserkerDotNet** [comment]
Thank you for pointing out missing license, I've added an MIT license.

I would love to see remoting as part of Unicorn 3 core. Initially I thought to make it as part of the Unicorn and submit a pull request, but then realized that it will be easier  for me to make it separate at least for version 1 ;)

I saw your experiments with Unicorn.Remoting but as I understood it's on "experimental" stage.
Will it be satisfactory if I adapt my "remoting" to Unicorn or you have specific ideas about that?

**Comment by kamsar** [comment]
Totally understand :)

The remoting I'm working on is a bit different: it's designed to allow a remote sitecore to act as a data source to sync from. Imagine, for example, copying production content down into a dev server by using  this sort of remoting. At present it's an experiment - and one not yet updated for v3.

In terms of adapting your remoting, that sounds fine with me. It looks like you're parsing the logs into an object model to send over as JSON? I think I'd prefer either direct text log transfer (maybe with polling for real time updates?), or else hooking directly into the logger classes to directly deliver the objects.

Unfortunately I haven't had a chance to actually try the VS part as I only have VS2015 lying around, and it's made for 2013. Is that an easy adaptation?

**Comment by BerserkerDotNet** [comment]
Wow, remoting sounds really cool. Currently we are using [RAZL](http://www.razl.net/) to do copying between environments.

VS 2015 is now supported :)

You are absolutely correct, currently I'm parsing log items on the server and sending them as serialized JSON to the client, it's not a problem to send them in plain text and parse on the client side. As for the "real time", right now I'm simply "interrogating" the server every 1 sec. for fresh updates (not really "real time" :( ). Initially I was thinking of using SignalR to do the real time part for me, but then thought that this might be like shooting a fly with a bazooka. I think simple "long polling" will work just fine in situation like this. I will start working on adapting my "remoting" to the Unicorn.

P.S. Judging from your last blog post: "No third party dependencies. (that means you, Ninject)", you don't want any third party dependency in Unicorn, am I right or there are exceptions?

**Comment by kamsar** [comment]
Yeah, probably right about SignalR. Polling is probably a good solution here for responsiveness.

I've got a streaming text console I use in some other projects, but partially reading a single long request (via C#) may be difficult; I used it streaming because Azure's load balancer drops HTTP connections that are idle for more than 4 minutes - like a non streaming Unicorn sync :-1: 

Third party dependencies are okay, as long as they don't fight with other folks' choice of the same dependency (like an IoC container). People got grumpy they needed Ninject when the rest of their site was on Autofac or SimpleInjector. If a hypothetical remote log reading API had dependencies, it might be worth splitting into its own NuGet package, but not necessarily.
